server/common/utils/helpers.js

```diff
@@ -536,6 +536,46 @@ const prepareRequestedItems = lists => userId => {
   return data.sort(reportDateComparator);
 };
 
+const getItemsForReport = async (listModel, user) => {
+  const { _id: userId, displayName, email: receiver } = user;
+  const reportData = { displayName, receiver };
+  const data = { requests: [], todos: [] };
+
+  const ownerLists = await listModel
+    .find({
+      ownerIds: { $in: [userId] },
+      isArchived: false,
+      'items.0': { $exists: true }
+    })
+    .lean()
+    .populate('cohortId', 'name')
+    .populate('items.authorId', 'displayName')
+    .exec();
+
+  const viewerLists = await listModel
+    .find({
+      viewersIds: { $in: [userId] },
+      isArchived: false,
+      'items.0': { $exists: true }
+    })
+    .lean()
+    .populate('cohortId', 'name')
+    .populate('items.authorId', 'displayName')
+    .exec();
+
+  if (ownerLists) {
+    data.todos = prepareTodosItems(ownerLists);
+  }
+
+  if (viewerLists) {
+    data.requests = prepareRequestedItems(viewerLists)(userId);
+  }
+
+  reportData.data = data;
```

Comment:
wouldnt' be easier to add displayName and receiver to data object on line 542 instead of creating second object, that contains `data` object?  Complicated.

`line 542:  const data = { requests: [], todos: [], displayName, receiver};`
`line 574  remove this line`
`line 576 return data`